src/routes/api/tlv/tlv-sources/blockscout-chains.ts

```diff
@@ -0,0 +1,79 @@
+import getOptionalEnvVar from '$lib/utils/get-optional-env-var/private';
+import { z } from 'zod';
+import type { TLVSourceFn } from '../types';
+
+const FILECOIN_BLOCKSCOUT_API_KEY = getOptionalEnvVar(
+  'FILECOIN_BLOCKSCOUT_API_KEY',
+  true,
+  "Filecoin TLV won't be considered for default explore page.",
+);
+
+const OPTIMISM_BLOCKSCOUT_API_KEY = getOptionalEnvVar(
+  'OPTIMISM_BLOCKSCOUT_API_KEY',
+  true,
+  "Optimism TLV won't be considered for default explore page.",
+);
+
+const blockscoutTokensResponseSchema = z.array(
+  z.object({
+    token: z.object({ address: z.string(), decimals: z.string() }),
+    value: z.string(),
+  }),
+);
+
+const apiKeys = {
+  optimism: OPTIMISM_BLOCKSCOUT_API_KEY,
+  filecoin: FILECOIN_BLOCKSCOUT_API_KEY,
+  metis: null,
+};
+
+const blockscoutInstanceUrls = {
+  optimism: 'https://optimism.blockscout.com',
+  filecoin: 'https://filecoin.blockscout.com',
+  metis: 'https://andromeda-explorer.metis.io',
+};
+
+async function fetchForNetwork(network: 'optimism' | 'filecoin' | 'metis', f: typeof fetch) {
+  const apiKey = apiKeys[network];
+
+  // undefined means env var is missing, null means it's not required, so we should continue
+  if (apiKey === undefined) return [];
+
+  const dripsTokenHoldingRes = await f(
+    `${blockscoutInstanceUrls[network]}/api/v2/addresses/0xd320F59F109c618b19707ea5C5F068020eA333B3/token-balances?apikey=${apiKey ?? ''}`,
```

Comment:
Extremely minor: I'd just add `0xd320F59F109c618b19707ea5C5F068020eA333B3` into a var just to know what this address is.